controllers/kogitoserverlessbuild_controller.go

```diff
@@ -57,41 +59,68 @@ type KogitoServerlessBuildReconciler struct {
+		log.Error(err, "Failed to get KogitoServerlessWorkflow")
+		return ctrl.Result{}, err
+	}
+
+	phase := build.Status.BuildPhase
+	if r.commonBuildConf.Data == nil {
+		r.commonBuildConf, err = utils.GetConfigMap(r.Client)
+	}
 
-		if err != nil || len(r.defaultBuildConf.Data[r.defaultBuildConf.Data[constants.DEFAULT_BUILDER_RESOURCE_NAME_KEY]]) == 0 {
-			return ctrl.Result{}, errors.NewNotFound(schema.GroupResource{
-				Resource: "ConfigMap",
-			}, "builder-config")
+	if err != nil || len(r.commonBuildConf.Data[r.commonBuildConf.Data[constants.DEFAULT_BUILDER_RESOURCE_NAME_KEY]]) == 0 {
+		return ctrl.Result{}, errors.NewNotFound(schema.GroupResource{
+			Resource: "ConfigMap",
+		}, "builder-config")
+	}
+	// Fetch the Platform build with the information we need for the build
+	pl, err := platform.GetActivePlatform(ctx, r.Client, req.Namespace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup
+			// logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
 		}
+		// Error reading the object - requeue the request.
+		log.Error(err, fmt.Sprintf("Error retrieving the active platfor. Workflow %s build cannot be performed!", build.Spec.WorkflowId))
+		return reconcile.Result{RequeueAfter: 60 * time.Second}, err
+	}
 
-		builder := builder.NewBuilder(ctx, r.defaultBuildConf)
+	customConfig, err := utils.GetCustomConfig(*pl)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Error retrieving the custom configuration from the platform %s in namespace %s. Workflow %s build cannot be performed", pl.Name, pl.Namespace, build.Spec.WorkflowId))
+		return reconcile.Result{RequeueAfter: 60 * time.Second}, err
+	}
 
-		if phase == api.BuildPhaseNone {
-			workflow, err := r.retrieveWorkflowFromCR(instance.Spec.WorkflowId, ctx, req)
-			if err == nil {
-				build, err := builder.ScheduleNewBuildWithContainerFile(instance.Spec.WorkflowId, workflow)
-				if err == nil {
-					manageStatusUpdate(ctx, build, instance, r, log)
-					return ctrl.Result{RequeueAfter: 1 * time.Second}, nil
-				}
-			}
-		} else if phase != api.BuildPhaseSucceeded && phase != api.BuildPhaseError && phase != api.BuildPhaseFailed {
-			cli, _ := clientr.NewClient(true)
-			build, err := builder.ReconcileBuild(&instance.Status.Builder, cli)
+	builder := builder.NewBuilderWithConfig(ctx, r.commonBuildConf, customConfig)
+
+	if phase == api.BuildPhaseNone {
+		workflow, err := r.retrieveWorkflowFromCR(build.Spec.WorkflowId, ctx, req)
+		if err == nil {
+			buildStatus, err := builder.ScheduleNewBuildWithContainerFile(build.Spec.WorkflowId, workflow)
 			if err == nil {
-				manageStatusUpdate(ctx, build, instance, r, log)
-				return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
+				manageStatusUpdate(ctx, buildStatus, build, r, log)
+				return ctrl.Result{RequeueAfter: 1 * time.Second}, nil
```

Comment:
isn't 1 second too small?

Reply:
yes, it is better to revert it to at least 5 seconds.